client/src/api/interest/index.ts

```diff
@@ -0,0 +1,47 @@
+import { IInterest } from 'ssw-common';
+
+import { Response } from '../types';
+import { buildEndpoint, get, post, put } from '../builders';
+
+import { InterestsResponse, InterestResponse } from './types';
+
+const INTEREST_ENDPOINT = '/interests';
+
+// Returns all of the interests from the database
+const getInterests = async (): Promise<Response<InterestsResponse>> => {
+  const url = buildEndpoint(INTEREST_ENDPOINT);
+  const failureMessage = 'Failed to get interests';
+
+  return await get(url, failureMessage);
+};
+
+// Returns a single interest from the database
+const getInterest = async (id: string): Promise<Response<InterestResponse>> => {
+  const url = buildEndpoint(INTEREST_ENDPOINT, id);
+  const failureMessage = 'Failed to get interest';
+
+  return await get(url, failureMessage);
+};
+
+// Creates a new interest in the database
+const createInterest = async (
+  interest: Partial<IInterest>,
+): Promise<Response<InterestResponse>> => {
+  const url = buildEndpoint(INTEREST_ENDPOINT);
+  const failureMessage = 'Failed to create interest';
+
+  return await post(url, { interest }, failureMessage);
+};
+
+// Updates an interest in the database
+const updateInterest = async (
+  id: string,
+  interest: Partial<IInterest>,
+): Promise<Response<InterestResponse>> => {
+  const url = buildEndpoint(INTEREST_ENDPOINT, id);
+  const failureMessage = 'Failed to update interest';
+
+  return await put(url, { interest }, failureMessage);
```

Comment:
See [this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJKvNCBnMyDeAUMicgPrAAmAXMrlKAOYDcxpIcAthLfU66WQIA9gBthUXmAYgWbEojDAAbj2QAjYWIhwQrAL6FCIkLmSQztAApwoSuKIA86TFBxgAfMgC8BeeSpaAHIggBp-Dm5gsP8RcUlkEPDBRRU1eFFsCHD9IA)

You can pass an entire IInterest into this as necessary, this just allows us to pass in partial versions of this as well. For PUT endpoints (consider PUT for a user). We don't want an entire IUser to be passed back because that would be too much information and unnecessary. We can just use Partial and pass the fields that we need back. If that means pass everything then so be it.

Reply:
oops github messed up the ordering of my comments but I meant the curly brace thing around interest for this comment. i understand the bracket thing now tho thanks!